extraConfigIsoBuilder.py

```diff
@@ -64,3 +66,91 @@ def ensure_image_is_built(h: host.Host, image_name: str, build_action: Callable[
         logger.info(f"Building '{image_name}'...")
         build_action()  # Execute the provided build logic
         logger.info(f"Successfully ensured image '{image_name}' is available.")
+
+
+def build_iso_builder_image_action(h: host.Host, iso_builder_image_name: str) -> None:
+    """
+    Action to build the ISO builder image. This will be passed to ensure_image_is_built.
+    Handles its own temporary directory and git cloning.
+    """
+    with tempfile.TemporaryDirectory() as tmpdir_path:
+        git_repo = os.path.join(tmpdir_path, "iso-builder")
+        common.git_repo_setup(git_repo, repo_wipe=True, url="https://github.com/SamD2021/ipu-rhel-iso-builder.git")
+        err, out, returncode = h.run(f"sudo podman build --security-opt label=type:unconfined_t " f"--platform linux/arm64 -t {iso_builder_image_name} {git_repo}")
+        if returncode:
+            logger.error_and_exit(f"Failed to build ISO builder image '{iso_builder_image_name}' with error: {err}")
+        logger.debug(f"ISO builder build output (stdout): {out}")
+        logger.debug(f"ISO builder build errors (stderr): {err}")
+
+
+def bootc_iso_builder(
+    h: host.Host,
+    name_of_final_iso: str,
+    secrets_path: str,
+    organization_id: str,
+    activation_key: str,
+    bootc_image_url: str,
+    image_builder_url: str,
+    rhel_version: str = "9.6",
+    input_iso: Optional[str] = None,
+    kickstart: Optional[str] = None,
+    kargs: Optional[str] = None,
+) -> None:
+    """
+    Uses temp directories to optionally build the Bootc iso builder and runs it as a cross arch container to build an arm64 image mode iso ready for DPU's
+    """
```

Comment:
More commentary to describe each step.
Break it into each own function with descriptive name.